### ISSUE TITLE
The train_test_split utility doesn't respect original order when shuffle=False

### ISSUE THREAD

**Issue by tobemo** [Lightning-AI/litdata#826]
## 🐛 Bug
The train_test_split code in litdata/utilities/train_test_split.py seems to rely on the alphabetical filename ordering of chunks, which is unreliable.
Stepping through the code I noticed the very first split contains files
curr_chunk_filename=['chunk-0-0.bin', 'chunk-1-0.bin', 'chunk-10-0.bin', 'chunk-2-0.bin', 'chunk-3-0.bin', 'chunk-4-0.bin', 'chunk-5-0.bin', 'chunk-6-0.bin']
on L91
Notice that chunk 10 snuck in there.

From testing it seems there are 2 failure paths:
- more than 10 chunks are written per worker
- more than 10 workers are used
As soon as a chunk has the name `chunk-10-x` or `chunk-x-10` it no longer sorts properly.

This is especially relevant when working with temporal data.

**Comment by deependujha** [comment]
Hi @tobemo, thanks for opening the issue. My findings are: `shuffle=False` doesn't work properly when using `num_workers>=10`.


but when `more than 10 chunks are written per worker & workers<10`, it works fine.

here's my sample code for testing:

```python
# optimize.py
import litdata as ld


def my_func(index):
    return (index, index**2)


if __name__ == "__main__":
    # The optimize function writes data in an optimized format
    ld.optimize(
        fn=my_func,  # the function applied to each input
        inputs=list(
            range(1000)
        ),  # the inputs to the function (here it's a list of numbers)
        output_dir="fast_data",  # optimized data is stored here
        num_workers=4,  # the number of workers on the same machine
        chunk_size=5,  # size of each chunk
        mode="overwrite",
    )
```

```python
# stream.py
import litdata as ld


def main():
    ds = ld.StreamingDataset("fast_data")
    for item in ds:
        print(item)
    print("\nSplitting the dataset into two halves:")
    splits = ld.train_test_split(ds, splits=[0.5, 0.5], shuffle=False)
    for i, split in enumerate(splits):
        print(f"Split {i}:")
        for item in split:
            print(f"\t{item}")


if __name__ == "__main__":
    main()
```

> when i increase `num_workers = 16`, output of `stream.py` isn't ordered.
